cql3/cql3_type.cc

```diff
@@ -207,9 +207,9 @@ class cql3_type::raw_ut : public raw {
             throw exceptions::invalid_request_exception(format("Unknown type {}", _name));
         }
         try {
-            auto&& type = user_types->get_type(_name.get_user_type_name());
-            if (!is_frozen()) {
-                throw exceptions::invalid_request_exception("Non-frozen User-Defined types are not supported, please use frozen<>");
+            data_type type = user_types->get_type(_name.get_user_type_name());
```

Comment:
This should be guarded with a cluster feature. Otherwise, a lucky client could connect to an new node, create a table with a thawed UDT, then try to access it via an old node and get crashes (at best).